src/controllers/routeActivation.ts

```diff
@@ -0,0 +1,105 @@
+const transformerTestModeEnabled = process.env.TRANSFORMER_TEST_MODE
+  ? process.env.TRANSFORMER_TEST_MODE.toLowerCase() === 'true'
+  : false;
+const areFunctionsEnabled = process.env.ENABLE_FUNCTIONS === 'false' ? false : true;
+const destinationFilterList = process.env.DESTINATION_FILTER_LIST?.toLocaleLowerCase();
+const sourceFilteList = process.env.SOURCE_FILTER_LIST?.toLocaleLowerCase();
+const deliveryFilterList = process.env.DESTINATION_DELIVERY_FILTER_LIST?.toLocaleLowerCase();
+
+export default class RouteActivationController {
+  private static executeActivationRule(ctx: Context, next: Next, condition: boolean) {
+    if (condition) {
+      ctx.status = 404;
+      ctx.body = 'RouteActivationController route is disabled';
+      return ctx;
+    } else {
+      return next();
+    }
+  }
+  public static isDestinationRouteActive(ctx: Context, next: Next) {
+    return RouteActivationController.executeActivationRule(ctx, next, !startDestTransformer);
+  }
+
+  public static isSourceRouteActive(ctx: Context, next: Next) {
+    return RouteActivationController.executeActivationRule(ctx, next, !startSourceTransformer);
+  }
+
+  public static isDeliveryRouteActive(ctx: Context, next: Next) {
+    return RouteActivationController.executeActivationRule(ctx, next, !transformerDelivery);
+  }
+
+  public static isDeliveryTestRouteActive(ctx: Context, next: Next) {
+    return RouteActivationController.executeActivationRule(ctx, next, !deliveryTestModeEnabled);
+  }
+
+  public static isUserTransformRouteActive(ctx: Context, next: Next) {
+    return RouteActivationController.executeActivationRule(ctx, next, !areFunctionsEnabled);
+  }
+
+  public static isUserTransformTestRouteActive(ctx: Context, next: Next) {
+    return RouteActivationController.executeActivationRule(ctx, next, !transformerTestModeEnabled);
+  }
+
+  public static destinationProcFilter(ctx: Context, next: Next) {
+    const { destination }: { destination: string } = ctx.params;
+    return RouteActivationController.executeActivationRule(
+      ctx,
+      next,
+      Array.isArray(destinationFilterList) &&
+        !destinationFilterList.split(',').includes(destination.toLowerCase()),
+    );
+  }
+
+  public static destinationRtFilter(ctx: Context, next: Next) {
+    const routerRequest = ctx.request.body as RouterTransformationRequest;
+    const destination = routerRequest.destType;
+    return RouteActivationController.executeActivationRule(
+      ctx,
+      next,
+      Array.isArray(destinationFilterList) &&
+        !destinationFilterList.split(',').includes(destination.toLowerCase()),
+    );
+  }
+
+  public static destinationBatchFilter(ctx: Context, next: Next) {
+    const routerRequest = ctx.request.body as RouterTransformationRequest;
+    const destination = routerRequest.destType;
+    return RouteActivationController.executeActivationRule(
+      ctx,
+      next,
+      Array.isArray(destinationFilterList) &&
```

Comment:
As `destinationFilterList` will always be a string(taken from env). `Array.isArray(destinationFilterList)` will always evaluate to `false`, so finally the condition would evaluate to `false`.
Even when you don't want a particular destination to use new routes, since the condition evaluates to `false` it will go via new routes

Please let me know if my analysis is right.

I would suggest to change to this
```suggestion
        !destinationFilterList.split(',').includes(destination.toLowerCase())
```

Reply:
Yes make sense

Reply:
Addressed